enhancements/storage/manila-csi-driver-operator.md

```diff
@@ -0,0 +1,230 @@
+Actions:
+
+- Create a new repo for manila csi driver image in OCP quay account.
+
+- Set up auto image building based on the changes in the cloud provider repo.
+
+#### Test the driver manually
+
+When all required components are built, we can manually deploy the driver with NFS backend and test how it works.
+Actions:
+
+- Manually deploy the driver on a cloud with Manila support
+
+- Configure it to use NFS backend
+
+- Go through the lifecycle of volume (create, update, mount, write data, delete, etc.)
+
+#### Write Manila CSI driver operator
+
+The operator should be able to create, configure and manage Manila CSI driver for Kubernetes and OpenShift. In other words, automate what has been done on the previous step.
+Actions:
+
+- Create a new repo in OpenShift’s github: github.com/openshift/manila-csi-driver-operator
+
+- Implement the driver, using [OpenShift Operator SDK](https://docs.openshift.com/container-platform/4.1/applications/operator_sdk/osdk-getting-started.html)
+
+#### Add automatic testing
+
+To make sure that everything works properly, we need the support of CI. If Manila is added to MOC, we just need to run a few tests in e2e-openstack pipeline. K8s already implements all necessary tests for CSI drivers, and they should run automatically if the testing system detects a CSI driver attached.
+Actions:
+
+- Run tests from https://github.com/openshift/origin/tree/master/vendor/k8s.io/kubernetes/test/e2e/storage/drivers against Manila CSI driver.
+
+#### Document how to use the driver
+
+Make sure users understand how to use the feature.
+
+Actions:
+
+- Write documentation for end users, that describes how they can create a PV in Manila.
+
+#### Build Manila CSI driver operator image and put it in Quay
+
+To start using Manila CSI operator in OpenShift we need to build an image and put it in the OCP Quay account along with other OCP images.
+Dockerfile should be automatically generated by the SDK.
+Actions:
+
+- Create a new repo for Manila CSI driver operator image in OCP quay account.
+
+- Set up auto image building based on the changes in the operator repo.
+
+### Risks and Mitigations
+
+#### No clouds were found with Manila support
+
+We won't be able to develop/test our work.
+
+Severity: blocker
+Likelihood: probably
+
+#### Manila CSI driver doesn’t work properly
+
+The driver has not been tested on either OSP or OCP. The team uses Devstack + K8s plugin for the development.
+
+Severity: medium-high
+Likelihood: probably
+
+When testing on industrial clouds, we may face challenges, like bugs, issues with compatibility and so on. We will need to address them accordingly.
+
+#### CSI tooling doesn’t work properly
+
+CSI tooling was introduced in OpenShift in 4.3 by the storage team. So far there are no CSI drivers available, and there can be some problems with operability.
+
+Severity: medium-high
+Likelihood: low
+
+#### The team will be busy with other things and will not be able to complete the tasks in time
```

Comment:
The time constraint is a Red Hat internal commitment, and is not part of the opensource enhancement. I suggest to remove this section.